enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,530 @@
+-A OVN-KUBE-EXTERNALIP -d 192.168.122.254/32 -p tcp -m tcp --dport 80 -j DNAT --to-destination 10.43.252.173:80
+...
+```
+
+Since ovnk picks up every element in `.status.loadBalancer.ingress` to create
+iptable rules, duplicates are ignored. Their rules already exist. However, for
+clarity reasons MicroShift should keep this list unique so that admins/users
+are not confused by it. MicroShift will not trigger warnings or errors if there
+are duplicate entries from the configuration (these include explicit duplicates
+or referring to the same IP in different ways, such as by their hostnames or
+interfaces).
+
+#### Firewalling ports
+Using LoadBalancer service will create special iptables rules with more
+precedence than those from firewalld. This means a `LoadBalancer` service is
+immune to such configurations. In order to firewall ports the user needs to
+take action by either creating `NetworkPolicy` resources or disabling the
+router.
+
+Disabling the router might be disruptive, as not all traffic may come from
+external sources. This procedure also requires a restart, which might not be
+desirable/possible in all situations.
+
+Using `NetworkPolicy` a user may control the traffic that is allowed into
+the cluster. By creating a deny-all policy the router will still remain
+operative for internal traffic, while rejecting all external traffic. An
+example follows:
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: block-router
+  namespace: openshift-ingress
+spec:
+  ingress:
+  - from:
+    - ipBlock:
+        cidr: 10.42.0.0/16
+  podSelector:
+    matchLabels:
+      ingresscontroller.operator.openshift.io/deployment-ingresscontroller: default
+  policyTypes:
+  - Ingress
+```
+Assuming the pod network is configured to use CIDR `10.42.0.0/16`, the above
+policy will block all traffic not coming from that network.
+
+Any of these methods is equivalent to using firewalld to block traffic, but
+using the CNI instead of a host level setting. This provides greater
+flexibility, as it is self contained within MicroShift APIs.
+
+If a user needs to block traffic to a port this needs a `NetworkPolicy`
+resource like the one shown above. This is NOT automatically done by
+MicroShift, user action is required.
+
+#### Exposing the router
+Relying on [LoadBalancer controller enhancement](loadbalancer-service-support.md)
+means the services are updated by this component. However, ovnk only turns IP
+addresses into iptables rules, it does not take Hostname, which is another
+valid field for the service status.
+
+Since the configuration uses NIC names the [service controller](loadbalancer-service-support.md)
+needs to translate them to their corresponding IP addresses, and then
+configure them in the service.
+
+Since the NIC names are present in the host, MicroShift will need to list he IP
+addresses from them and include them in the service `status` field. Due to the
```

Comment:
```suggestion
Since the NIC names are present in the host, MicroShift will need to list the NIC's IP
addresses and include them in the service `status` field. Due to the
```

Reply:
Done